src/cursor/abstract_cursor.ts

```diff
@@ -382,14 +389,7 @@ export abstract class AbstractCursor<
       return true;
     }
 
-    const doc = await next<TSchema>(this, { blocking: true, transform: false });
-
-    if (doc) {
-      this[kDocuments].unshift(doc);
-      return true;
-    }
-
-    return false;
+    return await next(this, { blocking: true, transform: false, hasNext: true });
```

Comment:
If you look through my commits I originally supported unshift so that hasNext worked without changes but it was more logic than it was worth to hold on to the unshifted document. I think this is a good improvement regardless of CursorResponse, I wouldn't mind merging it separately

Reply:
Here's the commit where I removed it: [`329ca55` (#4085)](https://github.com/mongodb/node-mongodb-native/pull/4085/commits/329ca55bcd58d5c8c798cf25936cf2c18fd71a30)

I have to buffer the unshifted documents to be returned by shift at a later point, in my view there is complexity & risk to maintaining this look-aside list that must be iterated first to ensure we don't repeat parsing effort and maintain the correct order, better to just not pull things out of a list.

Happy to change the name and also move this change out of this PR to keep it isolated. `checkNext`? `iterate: false`? `dequeue: false`?

Reply:
WIP https://github.com/mongodb/node-mongodb-native/pull/4093